src/app/util/attribute-storage.cpp

```diff
@@ -1329,11 +1346,32 @@ void emAfSaveAttributeToStorageIfNeeded(uint8_t * data, EndpointId endpoint, Clu
         return;
     }
 
-// On EZSP host we currently do not support this. We need to come up with some
-// callbacks.
-#ifndef EZSP_HOST
-    GENERATED_TOKEN_SAVER;
-#endif // EZSP_HOST
+    // TODO: Maybe we should have a separate constant for the size of the
+    // largest non-volatile attribute?
+    uint8_t allZeroData[ATTRIBUTE_LARGEST] = { 0 };
+    if (data == nullptr)
+    {
+        data = allZeroData;
+    }
+
+    size_t dataSize;
+    EmberAfAttributeType type = metadata->attributeType;
+    if (emberAfIsStringAttributeType(type))
+    {
+        dataSize = emberAfStringLength(data) + 1;
+    }
+    else if (emberAfIsLongStringAttributeType(type))
+    {
+        dataSize = emberAfLongStringLength(data) + 2;
+    }
+    else
+    {
+        dataSize = metadata->size;
+    }
+
+    auto * attrStorage = app::GetAttributePersistenceProvider();
```

Comment:
Must make sure this is not null

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/13600